consensus/istanbul/backend/engine.go

```diff
@@ -404,6 +405,15 @@ func (sb *Backend) Prepare(chain consensus.ChainReader, header *types.Header) er
 		header.Time = big.NewInt(time.Now().Unix())
 	}
 
+	parentExtra, err := types.ExtractIstanbulExtra(parent)
```

Comment:
@asaj Is this what you had in mind? In the `Prepare` we are modifying the header to also include the parent's info.